app/controllers/application_controller.rb

```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   include Pundit::Authorization
+
+  before_action :look_up_pages
+
+  private
+
+  def look_up_pages
+    @pages = Page.build_hierarchy Rails.root.join "_pages"
```

Comment:
This is a fine way to do things. Another (maybe more Railsy?) way would be to add `def pages` to `app/helpers/application_helper.rb` and call that method from the views.

If you do that, you'd want to memoize the value, which is really easy in ruby with just `@pages ||= Page.build...`